mindmeld/auto_annotator.py

```diff
@@ -835,5 +901,320 @@ def valid_entity_check(self, entity):
+        translator = kwargs.get("translator")
+        self.translator = TranslatorFactory().get_translator(translator)
+        self.en_annotator = kwargs.get("en_annotator") or SpacyAnnotator(
+            app_path=self.app_path,
+            language=ENGLISH_LANGUAGE_CODE,
+            locale=ENGLISH_US_LOCALE,
+        )
+
+    def parse(self, sentence, entity_types=None, **kwargs):
+        """
+        Args:
+            sentence (str): Sentence to detect entities.
+            entity_types (list): List of entity types to parse. If None, all
+                    possible entity types will be parsed.
+        Returns:
+            entities (list): List of entity dictionaries.
+        """
+        candidates = self.en_annotator.duckling.get_candidates_for_text(
+            sentence,
+            entity_types=entity_types,
+            language=self.language,
+            locale=self.locale,
+        )
+        en_sentence = self.translator.translate(  # pylint: disable=E1128
+            sentence, target_language=ENGLISH_LANGUAGE_CODE
+        )
+        en_entities = self.en_annotator.parse(en_sentence, entity_types=entity_types)
+        final_candidates = []
+        for entity in en_entities:
+            value_matched_candidates = []
+            for candidate in candidates:
+                if entity["dim"] != candidate["entity_type"]:
+                    continue
+                if entity["value"] == candidate["value"]:
+                    value_matched_candidates.append(candidate)
+                if value_matched_candidates:
+                    continue
+                if (
+                    self.translator.translate(
+                        entity["body"], target_language=self.language
+                    )
+                    == candidate["body"]
+                ):
+                    final_candidates.append(candidate)
+                    break
+            if value_matched_candidates:
+                final_candidates.append(
+                    max(value_matched_candidates, key=lambda x: len(x["body"]))
+                )
+        if entity_types:
+            final_candidates = [
+                e for e in final_candidates if e["entity_type"] in entity_types
+            ]
+        return [
+            duckling_item_to_entity_dict(candidate) for candidate in final_candidates
+        ]
+
+    @property
+    def supported_entity_types(self):  # pylint: disable=W0236
+        """
+        Returns:
+            supported_entity_types (list): List of supported entity types.
+        """
+        supported_entity_types = set(
+            self.en_annotator.supported_entity_types
+        ).intersection(DUCKLING_TO_SYS_ENTITY_MAPPINGS[self.language])
+        return list(supported_entity_types)
+
+
+class MultiLingualAnnotator(Annotator):
+    """Custom Annotator class used to generate annotations."""
+
+    def __init__(self, *args, **kwargs):
+        """Initializes a TranslationDucklingAnnotator.
```

Comment:
Initializes a MultiLingualAnnotator

Reply:
It would be helpful to write more detailed docs here on exactly what MLA is and how it is different from other annotators like TranslationAnnotator and NonTranslation annotator.